free-3.md

```diff
@@ -57,12 +68,14 @@ making language design decisions), as opposed to "systems" aspects of the
 project (e.g., implementing a complicated semantics that doesn't require a lot
 of language design)?_
 
+There doesn't seem to be that many language design decisions necessarily, more UI choices to be made to display the information in the best possible way. One difficult choice would be choosing which features to prioritize. 
 
 ### Scope
 _How big an idea is this? How ambitious is this project?_
-
+If a full GUI needs to be built, it would be quite an ambitious project - evident by how many people worked on the visualgo.net project. If we're ok with less of a UI, it could work on the terminal which would save time. I would need to figure out how to store recursive function calls on a stack, complete with its arguments. Then I would have to figure out how to generate a tree to visualize these recursive calls.  
```

Comment:
If you use Python there are modules to inspect the stack at runtime built into the language. Otherwise, if you implement a stack-based language then you just need to add hooks to be usable by your visualization framework.